### PR TITLE
Add concept of processing profiles

### LINKED ISSUE #3: Custom template parameters in batch mode
Lifted from https://github.com/plprobelab/ashby/pull/1#pullrequestreview-1422102519

The main goal is to allow production of plots that differ only in small ways that can be templated, e.g. a set of plots for website measurements, repeated for each website

We need to think of how this would work for batch mode though. There are a number of possibilities I can think of:

1. Do nothing and require each website page to have its own set of plotdefs with the website hardcoded
2. Extend the plotdef model to provide a list of values that should be iterated over when producing  plots. We could add a "variants" list, each variant would have a name (which is appended to the generated plot filename) and a set of template variables that sould be used for that variant (the website url for example). The main problem here is that I changed the templating to occur after the plotdef is read but before it is parsed. We would have to switch it back so that we could parse the plotdef, read the variants, iterate over them and inject the variables into the template for each iteration.
3. Do something similar to 2 but create a separate configuration that defines the variants. We could call these "processing profiles" and each plotdef could have a field that defines which profile it belongs to. 
4. Same as 3 but we don't change the plotdef. Instead we define everything in the config file. We have  "processing profiles" that define a list of plotdefs and a list of different sets of variables. Ashby would then perform each profile in turn, iterating over each set of variables and applying them to each plotdef in the profile. So for websites you would have a list of plotdefs that are needed by each website page and a list of different website urls that should be applied to each plotdef. This changes ashby's batch mode from simply iterating through a directory of plotdefs to iterating through a list of profiles, but the templating remains as it is: performed just after the plotdef is read. Would we need to define a profile for all the single-use plotdefs though?

### PR DESCRIPTION
**Pull request by dennis-tra**
These are my two processing profiles so far:

```yaml
directory: ../plotdefs
output: "{{.PlotDefFilename}}.json" # quotes are required
```

and

```yaml
directory: ../plotdefs-website
output: "{{.PlotDefFilename}}-{{ .Params.website }}-{{ .Params.type }}.json"
variants:
  - website: protocol.ai
    type: HTTP
  - website: protocol.ai
    type: KUBO
  - website: filecoin.ai
    type: HTTP
  - website: filecoin.io
    type: KUBO
```

### CODE REVIEW

File: batch.go [hunk 1]
```diff
@@ -181,133 +182,177 @@ func Batch(cc *cli.Context) error {
...
+			return fmt.Errorf("processing plot definitions: %w", err)
+		}
+	}
+
+	return nil
+}
 
-	org := new(Organizer)
-	absOutDir, err := filepath.Abs(batchOpts.outDir)
+func (p *ProcessingProfile) processPlotDefs(ctx context.Context, cfg *PlotConfig) error {
```
Comment (dennis-tra) [inline batch.go:201]: I've attached this method to the `ProcessingProfile` struct.  Not sure about this 🤷‍♂️ made it easy to access a few parameters

File: organize.go [hunk 2]
```diff
@@ -1,8 +1,10 @@
 package main
 
 import (
+	"bytes"
 	"errors"
 	"fmt"
+	"html/template"
```
Comment (iand) [inline organize.go]: any reason to choose html/template instead of text/template?

File: batch.go [hunk 3]
```diff
@@ -44,7 +45,7 @@ var batchCommand = &cli.Command{
 		&cli.StringFlag{
 			Name:        "in",
 			Required:    true,
-			Usage:       "Path of directory containing plot definitions.",
+			Usage:       "Path of directory containing processing profile definitions.",
```
Comment (iand) [inline batch.go]: We could put the profiles into a single profiles.yaml file in the config folder (already specified with `--conf`). Remove the `--in` option completely and let it be handled with the config.

File: organize.go (hunk 2 above)
Comment (dennis-tra) [reply organize.go]: oops, auto-import

File: batch.go (hunk 3 above)
Comment (dennis-tra) [reply batch.go]: That's a good idea 👍

Comment (dennis-tra) [reply batch.go]: done 👍

**Review by iand — APPROVED** [review]

**Pull request merged**